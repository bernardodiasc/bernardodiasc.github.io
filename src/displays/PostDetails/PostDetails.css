.PostDetails {
  font-size: 1rem;
}

.PostDetails__title {
  color: var(--heading-color);
  font-size: var(--heading-size);
  margin: 0 0 0.5em;
}

.PostDetails__meta {
  color: var(--meta-color);
  font-size: var(--meta-size);
  margin: 0 0 1em;
}
