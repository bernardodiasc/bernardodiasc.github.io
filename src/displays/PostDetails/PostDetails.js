import React, { PureComponent } from 'react'
import './PostDetails.css'

import PostHeader from 'displays/PostHeader'
import TextBlock from 'displays/TextBlock'
import MarkdownRenderer from 'displays/MarkdownRenderer'
import Share from 'displays/Share'

class PostDetails extends PureComponent {
  static defaultProps = {
    post: {
      handle: '',
      title: '',
      date: '',
      category: {
        handle: '',
        title: '',
        icon: '',
      },
      body: '',
    },
  }

  render() {
    const {
      handle,
      title,
      date,
      category,
      body,
    } = this.props.post
    const url = handle ? [
      handle.substring(0, 4),
      handle.substring(5, 7),
      handle.substring(8, 10),
      handle.substring(11),
    ].join('/') : ''

    return (
      <div className="PostDetails">
        <PostHeader
          mode="details"
          title={title}
          date={date}
          category={category}
        />
        {title && url && (
          <Share title={title} url={`${window.location.hostname}/${url}`} />
        )}
        <TextBlock>
          <MarkdownRenderer text={body} />
        </TextBlock>
      </div>
    )
  }
}

export default PostDetails
