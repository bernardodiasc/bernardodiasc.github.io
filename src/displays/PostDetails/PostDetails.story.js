import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import Docs from 'stories/containers/Docs'

import PostDetails from './PostDetails'
import PostDetailsReadme from './README.md'
storiesOf('Displays/PostDetails', module)
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{PostDetailsReadme}</Docs>)
  .add('Example', () => <PostDetails />)
  .add('Minimal render', () => <PostDetails />)
