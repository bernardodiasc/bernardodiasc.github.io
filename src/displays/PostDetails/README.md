# PostDetails

## How to use

```
import PostDetails from 'displays/PostDetails'
```

```
<PostDetails />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
