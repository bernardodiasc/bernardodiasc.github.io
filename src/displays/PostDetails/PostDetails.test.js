import React from 'react'
import { shallow } from 'enzyme'

import PostDetails from './PostDetails.js'

it('renders without crashing', () => {
  const component = shallow(<PostDetails />)
  expect(component).toMatchSnapshot()
})
