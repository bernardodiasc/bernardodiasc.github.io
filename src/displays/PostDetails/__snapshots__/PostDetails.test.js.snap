// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="PostDetails"
>
  <h1
    className="PostDetails__title"
  />
  <p
    className="PostDetails__meta"
  />
  <TextBlock />
</div>
`;
